### DISCUSSION TITLE
Fetch translations from API?

### DISCUSSION THREAD

**Discussion by Nobi322** [fluent-vue/fluent-vue#783] [Q&A] [answered]
Hey!
Is there a way to fetch translations from API? That's would be a nice addition

P.S.
Thanks for nice plugin! Слава Україні!

**Comment by Demivan** [comment, marked as answer]
Hi. It is quite easy to do in userland already.

https://fluent-vue.demivan.me/howto/importing-ftl-files.html combine this with fetching.

Pseudocode:
```js
import { FluentBundle, FluentResource } from '@fluent/bundle'

function changeLocale(locale) {
  // Or any other way of fetching messages
  const messages = await fetch(`/translations/${locale}.ftl`)

  const bundle = new FluentBundle(locale)
  bundle.addResource(new FluentResource(messages))

  fluent.bundles = [bundle]
}
```

Or you have something else in mind?

P.S.
Thanks for kind words. Героям Слава!

**Reply by Nobi322** [reply]
I will try that, thanks!

**Reply by Demivan** [reply]
What are you using for managing translations? Why do you load them from API?

**Reply by Nobi322** [reply]
I have a rest API with admin panel that manage translations. So I need to fetch them to keep updated on FE(with some cache period).

**Reply by Demivan** [reply]
Have you managed to set it up?

By the way check this example:
https://github.com/fluent-vue/examples/tree/main/vue-3-vite-external-translations

It uses new package - [unplugin-fluent-vue](https://github.com/fluent-vue/unplugin-fluent-vue) to import translations from external `.ftl` files. This allows to manage these files using external tools - for example [Pontoon](https://github.com/mozilla/pontoon)
